src/test/java/org/broadinstitute/hellbender/tools/spark/sv/FindBadGenomicKmersSparkUnitTest.java

```diff
@@ -0,0 +1,95 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceMultiSource;
+import org.broadinstitute.hellbender.engine.datasources.ReferenceWindowFunctions;
+import org.broadinstitute.hellbender.engine.spark.SparkContextFactory;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * Unit tests for FindBadGenomicKmersSpark
+ */
+public class FindBadGenomicKmersSparkUnitTest extends BaseTest {
+
+    private static final int KMER_SIZE = SVConstants.KMER_SIZE;
+    private static final String REFERENCE_FILE_NAME = hg19MiniReference;
+
+    @Test(groups = "spark")
+    public void badKmersTest() throws IOException {
+        final File refSequenceFile = File.createTempFile("genomic_sequence", "txt");
+        refSequenceFile.deleteOnExit();
+
+        final String polyA = StringUtils.repeat('A', KMER_SIZE);
+        final String polyC = StringUtils.repeat('C', KMER_SIZE);
+        final String polyT = StringUtils.repeat('T', KMER_SIZE);
+
+        try ( PrintWriter writer = new PrintWriter(refSequenceFile) ) {
+            long nTimes = FindBadGenomicKmersSpark.MAX_KMER_FREQ;
+            // write polyA and polyC the max number of times possible while evading the trigger
+            while ( nTimes-- > 0L ) {
+                writer.println(polyA);
+                writer.println(polyC);
+            }
+            // tip polyA over the edge by writing its reverse complement
+            writer.println(polyT);
+        }
+
+        final JavaSparkContext ctx = SparkContextFactory.getTestSparkContext();
+        final List<SVKmer> badKmers = FindBadGenomicKmersSpark.processRefFile(ctx, refSequenceFile);
+
+        // should have just one bad kmer:  polyA
+        Assert.assertEquals(badKmers.size(), 1);
+        Assert.assertEquals(badKmers.get(0).toString(KMER_SIZE), polyA);
+
+        if ( !refSequenceFile.delete() )
```

Comment:
Style preference on my part:  Insofar as possible each unit test ought to take care to try to restore the world to the _status quo ante_.  The `deleteOnExit` is necessary in case we blow out of the test on an exception.  But we oughtn't to rely on that for the normal code flow.  We ought to clean up as the test ends, rather than waiting for exit, so that our having created this file in this test can't possibly interfere with other tests.